### PR TITLE
Create stlink PKGBUILD

### PR DESCRIPTION
**Pull request by Fabien-Chouteau**
Firmware programmer for STM32 STLINK v1/v2 protocol

### CODE REVIEW

**Comment by lazka** [comment]
ha, I remember having to use a proprietary thing on Windows to work with this 10 years ago :D nice!

File: mingw-w64-stlink/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+             "${MINGW_PACKAGE_PREFIX}-gcc")
+source=(${_realname}-${pkgver}.zip::https://github.com/stlink-org/stlink/archive/v$pkgver.zip)
+sha256sums=('859af69fae8dc0751178c93a0558c55a99f4c57c17287ca32c8344021d56f170')
+
+build() {
+  [[ -d "${srcdir}/build-${MINGW_CHOST}" ]] && rm -rf "${srcdir}/build-${MINGW_CHOST}"
+  mkdir -p "${srcdir}/build-${MINGW_CHOST}"
+  cd "${srcdir}/build-${MINGW_CHOST}"
+
+  ${MINGW_PREFIX}/bin/cmake \
```
Comment (lazka) [inline mingw-w64-stlink/PKGBUILD:23]: Shouldn't there be a `MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \` before?

File: mingw-w64-stlink/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    -G"MSYS Makefiles" \
```
Comment (lazka) [inline mingw-w64-stlink/PKGBUILD:24]: can we use ninja?

see https://github.com/msys2/MINGW-packages/commit/a59d7a471b9d7d08d60b73656da064b896dda078 for a similar change

File: mingw-w64-stlink/PKGBUILD (hunk 1 above)
Comment (Fabien-Chouteau) [reply mingw-w64-stlink/PKGBUILD:23]: I don't know, I very new to the msys2 ecosystem. I can try with it if you want.

File: mingw-w64-stlink/PKGBUILD (hunk 2 above)
Comment (Fabien-Chouteau) [reply mingw-w64-stlink/PKGBUILD:24]: I can give it a try.

**Comment by Fabien-Chouteau** [comment]
@lazka new version up.

**Comment by lazka** [comment]
thanks!

**Comment by Fabien-Chouteau** [comment]
Thank you!

**Pull request merged**